gobblin-modules/gobblin-sql/src/test/java/org/apache/gobblin/writer/JdbcPublisherTest.java

```diff
@@ -87,6 +87,22 @@ public void testPublish() throws IOException, SQLException {
     inOrder.verify(workUnitState, times(1)).setWorkingState(WorkUnitState.WorkingState.COMMITTED);
     inOrder.verify(conn, times(1)).commit();
     inOrder.verify(conn, times(1)).close();
+  }
+
+  @Test(dependsOnMethods = { "testPublish" })
```

Comment:
Let me know if you have suggestions for improving this unit test

Reply:
I can't confirm that it would actually exercise any of the mysql-specific code you modified... does it?

I didn't see you update the `JdbcPublisher` class (above), so perhaps the better unit test to augment would be `MySqlBufferedInserterTest`?

Reply:
It's a bit hard to tell but these tests use `copyTable(database, stagingTable, destinationTable)` which I modified above. However adding a unit test in `MySqlBufferedInserterTest` would be a more direct approach to achieving this.

Reply:
update: added a better test in `MySqlBufferedInserterTest ` that checks for a `REPLACE` statement versus `INSERT`